crates/runtime/src/actor/component/mod.rs

```diff
@@ -418,17 +418,37 @@ impl Instance {
 
     /// Instantiates and returns [`GuestBindings`] if exported by the [`Instance`].
     async fn as_guest_bindings(&mut self) -> anyhow::Result<GuestBindings> {
-        if let Ok((bindings, _)) =
-            guest_bindings::Guest::instantiate_async(&mut self.store, &self.component, &self.linker)
-                .await
+        let mut errors = Vec::new();
+
+        // Attempt to instantiate using guest bindings
+        match guest_bindings::Guest::instantiate_async(
+            &mut self.store,
+            &self.component,
+            &self.linker,
+        )
+        .await
         {
-            Ok(GuestBindings::Interface(bindings))
-        } else {
-            let (bindings, _) = Command::instantiate_async(&mut self.store, &self.component, &self.linker).await.context(
-                    "failed to instantiate either `wasmcloud::bus/guest` interface or get `wasi:command/command`",
-                )?;
-            Ok(GuestBindings::Command(bindings))
-        }
+            Ok((bindings, _)) => return Ok(GuestBindings::Interface(bindings)),
+            Err(e) => errors.push(e),
+        };
+
+        // Attempt to instantiate using only bindings available in command
+        match Command::instantiate_async(&mut self.store, &self.component, &self.linker).await {
+            Ok((bindings, _)) => return Ok(GuestBindings::Command(bindings)),
```

Comment:
If we bail in the other match case, 
```suggestion
            Ok((bindings, _)) => Ok(GuestBindings::Command(bindings)),
```